gherkin/java/src/main/java/io/cucumber/gherkin/GherkinDialect.java

```diff
@@ -75,6 +105,10 @@ public List<String> getStepKeywords() {
         return stepKeywords;
     }
 
+    public Map<String, StepKeywordType> getStepKeywordsTypes() {
```

Comment:
Because the only caller immediate calls `.get(keyword)` I feel this should be something more like:

```
public StepKeywordType getStepKeywordType(String keyword)
```

I'm a little unsure about the UNKNOWN result - it might be better to return a list of possibilities? It depends how you want to consume it in cucumber

Reply:
Yes. I'm happy to change the code to return all the possibilitiies; I was looking at the use of the Dialect in the code base. It's really a very low in the stack of abstractions and a building block nobody saw fit to use for anything but i18n-ing the token matcher. which is why I coded the output to the currently-known use-case. And since this call will happen on every matched keyword, I was thinking it's rather inefficient to conclude every time that a keyword maps to multiple types, only to conclude *then* that the keyword type must be set to UNKNOWN.

Reply:
This is now implemented as a list of reverse mappings.